pkg/controllers/disruption/consolidation.go

```diff
@@ -192,6 +193,52 @@ func (c *consolidation) computeConsolidation(ctx context.Context, candidates ...
 		results.NewNodeClaims[0].Requirements.Add(scheduling.NewRequirement(v1beta1.CapacityTypeLabelKey, v1.NodeSelectorOpIn, v1beta1.CapacityTypeSpot))
 	}
 
+	results.NewNodeClaims[0].NodeClaimTemplate.InstanceTypeOptions = instanceTypesOptionsFilteredByPrice
+	return Command{
+		candidates:   candidates,
+		replacements: results.NewNodeClaims,
+	}, nil
+}
+
+// Compute command to execute spot-to-spot consolidation if:
+// 1. The SpotToSpotConsolidation feature flag is set to true.
+// 2. There are atleast 15 cheapest instance type replacement options to consolidate.
+// 3. The current candidate is NOT part of the first 15 cheapest instance types inorder to avoid repeated consolidation for single-node consolidation.
+func (c *consolidation) computeSpotToSpotConsolidation(ctx context.Context, candidates []*Candidate,
+	results *pscheduling.Results, instanceTypesOptionsFilteredByPrice []*cloudprovider.InstanceType) (Command, error) {
+
+	if !options.FromContext(ctx).FeatureGates.SpotToSpotConsolidation {
+		if len(candidates) == 1 {
+			// Spot consolidation is turned off.
+			c.recorder.Publish(disruptionevents.Unconsolidatable(candidates[0].Node, candidates[0].NodeClaim, "SpotToSpotConsolidation is disabled, can't replace a spot node with a spot node")...)
+		}
+		return Command{}, nil
+	}
+
+	// Allow single-node/multi-node consolidation only if there are minimum number of possible InstanceTypes for a NodeClaim
+	if len(results.NewNodeClaims[0].NodeClaimTemplate.InstanceTypeOptions) < MinInstanceTypesForSpotToSpotConsolidation {
+		// InstanceType flexibility is not good enough to replace spot node.
+		c.recorder.Publish(disruptionevents.Unconsolidatable(candidates[0].Node, candidates[0].NodeClaim, fmt.Sprintf("SpotToSpotConsolidation requires %d instance type replacement options to consolidate",
```

Comment:
You probably need a length check here `if len(candidates) == 1` to ensure that we aren't firing this event a bunch when we are checking for the possibility of a multi-node consolidation

Reply:
Nick, we should be having this check for multi-node spot consolidation too. But I think Jonathan wants to publish the events only for single-node consolidation because multi-node consolidation would trigger a bunch of events causing flooding to customers. 

This has been the considered today with respect to the code here too: https://github.com/kubernetes-sigs/karpenter/blob/main/pkg/controllers/disruption/consolidation.go#L180-L182

We can discuss more on what is the right approach here.

Reply:
> But I think Jonathan wants to publish the events only for single-node consolidation because multi-node consolidation would trigger a bunch of events causing flooding to customers

Yeah, if we do events on multi-node consolidation, then we will get (in the worst-case): 100 + 50 + 25 + 13 + 7 + 4 + 2 events since we would fire each time we do the evaluation in the multi-node consolidation loop. Since we would fire this event anyways in a single node consolidation loop, I don't think there's value (it's perhaps worse from a performance perspective) if we fire during multi-node consolidation

Reply:
I was thinking it might be more valuable to have the events be emitted for multi-node consolidation, even if it's just for the first node, but I'm thinking more now that it might be more confusing to see an event that was emitted with other omitted information.

Reply:
Conclusion: We are emitting this event only for single-node consolidation. Let me know if guys think otherwise.